runtime/util_core/j9argscan.c

```diff
@@ -385,6 +385,104 @@ scan_hex_caseflag_u64(char **scan_start, BOOLEAN uppercaseAllowed, uint64_t* res
 	return bits;
 }
 
+/**
+ * Scan the next unsigned number off of the argument string, and parses for memory sizes.
+ * Specify the size in GiBs, MiBs, or KiBs (with G,g,M,m,K,k suffixes) or in bytes (no suffix).
+ * @param[in] scan_start The string to be scanned
+ * @param[out] result The result
+ * @return returns 0 on success, 1 if the argument string is not a number, or 2 if overflow occurs.
+ */
+uintptr_t
+scan_u64_memory_size(char **scan_start, uint64_t* result)
+{
+	uintptr_t rc = scan_u64(scan_start, result);
+
+	/* Report any errors from scan */
+	if (0 != rc) {
+		return rc;
+	}
+
+	if(try_scan(scan_start, "T") || try_scan(scan_start, "t")) {
+		if (0 != *result) {
```

Comment:
This check is not required. In `scan_udata_memory_size()` it is necessary to handle exceptional case `0T` which is technically valid even on 32 but platform for uintptr_t size. However we have `uint64_t` here so there is no exception necessary.

Reply:
Makes sense, thanks for bringing that up. 
See: 1707149cde281821202101158513bd7bbfa20459